opm/parser/eclipse/EclipseState/Schedule/Well.cpp

```diff
@@ -303,11 +526,109 @@ namespace Opm {
         }
         return wellNameInPattern;
     }
+    bool Well::isMultiSegment() const {
+        return m_is_multi_segment;
+    }
+
+    void Well::setMultiSegment(const bool is_multi_segment) {
+        m_is_multi_segment = is_multi_segment;
+    }
+
+    SegmentSetConstPtr Well::getSegmentSet(size_t time_step) const {
+        return m_segmentset->get(time_step);
+    }
+
 
     bool Well::getAllowCrossFlow() const {
         return m_allowCrossFlow;
     }
 
+    void Well::processCOMPSEGS(size_t time_step, std::vector<CompsegsPtr>& compsegs) {
```

Comment:
I am struggling to understand what happens here (please: clearer code - not comments!). First of all - what is the main purpose of this method? Further questions:

1. What is the interaction between the input `std::vector<CompsegsPtr>& compsegs` and the state field `m_segmentset`?
2. Method seems to contain roughly two parts: 

   1. go through all the segments and update `segment_number` and `center_depth`.
   2. update the completions base on the updates segments.

   Could these two main tasks be split in two methods?